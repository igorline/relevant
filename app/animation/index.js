import React, {
  AppRegistry,
  Component,
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  ScrollView,
  Picker,
  ListView,
  Animated,
  Easing,
  RecyclerViewBackedScrollView
} from 'react-native';
import { globalStyles, fullWidth, fullHeight } from '../styles/global';

module.exports = {
  investAni: function(parent) {
    var styles = {...globalStyles};
    for (var i = 0; i < 21; i++) {
      var values = {
        x: new Animated.Value(0),
        y: new Animated.Value(0),
        scale: new Animated.Value(1.5),
        opacity: new Animated.Value(1)
      };
      parent.state.investAni.push(<Animated.Text style={[styles.aniMoney, {transform: [{translateX: values.x}, {scale: values.scale}, {translateY: values.y}], opacity: values.opacity}]}>ðŸ’µ</Animated.Text>);

      Animated.timing(values.x, {
        toValue: -(fullWidth/2),
        delay: i*100,
        duration: 500,
        easing: Easing.linear
      }).start();

      Animated.timing(values.y, {
        toValue: fullHeight*0.1,
        delay: i*100,
        duration: 500,
        easing: Easing.linear
      }).start();

      Animated.timing(values.opacity, {
        toValue: 0,
        delay: i*100,
        duration: 500,
        easing: Easing.linear
      }).start();

      Animated.timing(values.scale, {
        toValue: 5,
        delay: i*100,
        duration: 500,
        easing: Easing.linear
      }).start();
    }
    parent.setState({})
  }
};