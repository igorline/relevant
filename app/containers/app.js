'use strict';

import React, {
  AppRegistry,
  Component,
  StyleSheet,
  Text,
  View,
  TextInput
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as authActions from '../actions/authActions';
var {Router, routerReducer, Route, Container, Animations, Schema} = require('react-native-redux-router');
var {NavBar, NavBarModal} = require('../components/NavBar');
import Auth from './auth';
import Import from './import';
import Profile from './profile';

class Application extends Component {
  render() {
    return (

    <View style={{flex:1}} >
                <View style={{position:'absolute',left:0,right:0,top:0,bottom:0,backgroundColor:'white'}}/>
                <Router>
                    <Schema name="modal" sceneConfig={Animations.FlatFloatFromBottom} navBar={NavBarModal}/>
                    <Schema name="default" sceneConfig={Animations.FlatFloatFromRight} navBar={NavBar}/>
                    <Schema name="withoutAnimation" navBar={NavBar}/>
                    <Schema name="tab" navBar={NavBar}/>

                  <Route name="Auth" component={Auth}  hideNavBar={true} type="replace" initial={true} title="Home" />
                  <Route name="SignUp" component={Auth} hideNavBar={true} type="replace" title="Sign up" />
                  <Route name="LogIn" component={Auth} hideNavBar={true}  type="replace" title="Log in" />
                  <Route name="Import" component={Import} hideNavBar={true}  type="replace" title="Import" />
                  <Route name="Profile" component={Profile} hideNavBar={true}  type="replace" title="Profile" />
                </Router>

            </View>

    );
  }
}

export default Application

