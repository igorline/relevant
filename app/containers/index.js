import React, { Component } from 'react-native';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import * as reducers from '../reducers';
import Application from './app';

const reducer = combineReducers(reducers);
let store = applyMiddleware(thunk)(createStore)(reducer);
// let store = createStore(combineReducers(reducers));

export default class AppContainer extends Component {
  render() {
    return (
      <Provider store={store}>
         <Application />
      </Provider>
    );
  }
}