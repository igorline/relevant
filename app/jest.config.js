// TODO - move this stuff to setup file?
process.env.WEB = 'true';
process.env.API_SERVER = '';
process.env.BROWSER = true;
process.env.NETWORK_NUMBER = 99;

module.exports = {
  displayName: 'App',
  rootDir: '../',
  testPathIgnorePatterns: ['<rootDir>/node_modules/', '<rootDir>/server/'],
  preset: '<rootDir>node_modules/react-native-web',
  testEnvironment: 'jsdom',
  transform: {
    '^.+\\.(js|jsx)$': '<rootDir>/node_modules/babel-jest',
    '^[./a-zA-Z0-9$_-]+\\.(bmp|gif|jpg|jpeg|png|psd|svg|webp)$':
      '<rootDir>/__mocks__/fileTransformer.js'
  },
  transformIgnorePatterns: ['[/\\\\]node_modules[/\\\\].+\\.(js|jsx)$'],
  // for some react-native-web aliasing is causing a
  // console / 'canvas' error - this mocks canvas as a solution
  setupFilesAfterEnv: ['<rootDir>/app/jest.setup.js'],
  setupFiles: ['jest-canvas-mock', '<rootDir>/node_modules/regenerator-runtime/runtime']
};
