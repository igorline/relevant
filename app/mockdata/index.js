import * as activity from 'app/mockdata/activity';
import * as auth from 'app/mockdata/auth';
import * as community from 'app/mockdata/community';
import * as earnings from 'app/mockdata/earnings';
import * as post from 'app/mockdata/post';
import * as user from 'app/mockdata/user';
import * as twitter from 'app/mockdata/twitter';

export { activity, auth, community, earnings, post, user, twitter };
