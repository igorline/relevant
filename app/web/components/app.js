import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Drizzle } from 'drizzle';
// import { DrizzleProvider, drizzleConnect } from 'drizzle-react';
import Header from './common/header.component';
import AppHeader from './common/appHeader.component';
import * as routerActions from 'react-router-redux';

// import Footer from './common/footer.component';
import * as authActions from '../../actions/auth.actions';
import RelevantCoin from '../../contracts/RelevantCoin.json';
import AddEthAddress from './wallet/AddEthAddress';
import LoginModal from './auth/loginModal.component';
import AuthContainer from './auth/auth.container';
import Sidebar from './common/sidebar.component';
import EthTools from './ethTools/tools.container';
import Modal from './common/modal';
import CreatePost from './createPost/createPost.container';
import Eth from './ethTools/eth.context';

if (process.env.BROWSER === true) {
  console.log('BROWSER, import css');
  require('./index.css');
  require('./fonts.css');
  require('./splash/splash.css');
}


let options = {
  contracts: [
    RelevantCoin
  ],
  events: {},
  polls: {
    blocks: 300,
    accounts: 300,
  },
  networkId: 4,
  web3: {
    useMetamask: true,
    block: false,
    fallback: {
      type: 'ws',
      url: 'wss://rinkeby.infura.io/_ws'
    }
  }
};

const ThemeContext = React.createContext('light');

class App extends Component {
  static contextTypes = {
    store: PropTypes.object
  }

  state = {
    openLoginModal: false
  }

  componentDidMount() {
    // document.body.classList.remove('loading')
    this.props.actions.getUser();
    new Drizzle(options, this.context.store);
  }

  componentDidUpdate(prevProps) {
    if (this.props.location.pathname !== prevProps.location.pathname) {
      window.scrollTo(0, 0);
    }
    if (prevProps.auth.user !== this.props.auth.user) {
      let userId = this.props.auth.user ? this.props.auth.user._id : null;
      this.props.actions.userToSocket(userId);
    }
  }

  toggleLogin() {
    this.setState({ openLoginModal: !this.state.openLoginModal });
  }

  closeModal() {
    this.props.actions.push(this.props.location.pathname);
  }

  render() {
    const location = this.props.location;

    let mobileEl = (
      <div className="mobileSplash">
        <h1>Relevant browser version doesn't currently support mobile devices</h1>
        <p>Please download a dedicated mobile app:</p>
        <p>
          <a
            href="https://itunes.apple.com/us/app/relevant-a-social-news-reader/id1173025051?mt=8"
            target="_blank"
          >
            <img
              alt="iOS App Store"
              src="https://relevant.community/img/appstore.png"
            />
          </a>

          &nbsp;&nbsp;&nbsp;&nbsp;
          <a
            href="https://play.google.com/store/apps/details?id=com.relevantnative&amp;pcampaignid=MKT-Other-global-all-co-prtnr-py-PartBadge-Mar2515-1"
            target="_blank"
          >
            <img
              alt="Google Play Store"
              src="https://relevant.community/img/googleplaystore.png"
            />
          </a>
        </p>
      </div>
    );

    let header =
      <AppHeader
        location={location}
        params={this.props.params}
        toggleLogin={this.toggleLogin.bind(this)}
      />;
    if (location.pathname === '/') {
      header = <Header
        params={this.props.params}
        toggleLogin={this.toggleLogin.bind(this)}
      />;
      mobileEl = null;
    }

    let user = this.props.user;
    let temp = user && user.role === 'temp';
    const create = location.hash === '#newpost';
    const connectAccount = location.hash === '#connectAccount';

    return (
      <main>
        <EthTools>
          {header}
          {mobileEl}
          <div style={{ display: 'flex', width: '100%' }}>
            {this.props.children}
          </div>
          <AuthContainer
            toggleLogin={this.toggleLogin.bind(this)}
            open={this.state.openLoginModal || temp}
            modal
            {...this.props}
          />
          <Eth.Consumer>
            {wallet => <AddEthAddress
              connectAccount={connectAccount}
              closeModal={this.closeModal.bind(this)}
              {...this.props}
              {...wallet}
            />}
          </Eth.Consumer>
        </EthTools>
        <Modal
          title="New Post"
          visible={create}
          className="createPostModal"
          close={() => this.props.actions.push(location.pathname)}
        >
          <CreatePost modal />
        </Modal>
      </main>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  auth: state.auth
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    ...routerActions,
    ...authActions,
  }, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
