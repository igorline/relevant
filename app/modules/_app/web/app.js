import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import Header from 'modules/navigation/web/header.component';
import AuthContainer from 'modules/auth/web/auth.container';
import * as navigationActions from 'modules/navigation/navigation.actions';
import * as authActions from 'modules/auth/auth.actions';
import { getEarnings } from 'modules/wallet/earnings.actions';
import { getCommunities } from 'modules/community/community.actions';
import AddEthAddress from 'modules/wallet/web/AddEthAddress';
import Modal from 'modules/ui/web/modal';
import CreatePost from 'modules/createPost/createPost.container';
import EthTools from 'modules/web_ethTools/tools.container';
import Eth from 'modules/web_ethTools/eth.context';
import { ToastContainer } from 'react-toastify';
import { GlobalStyle } from 'app/styles';
import * as modals from 'modules/ui/modals';

if (process.env.BROWSER === true) {
  require('app/styles/index.css');
  require('app/styles/fonts.css');
  require('modules/web_splash/splash.css');
  require('react-toastify/dist/ReactToastify.css');
}

class App extends Component {
  static propTypes = {
    auth: PropTypes.object,
    actions: PropTypes.object,
    match: PropTypes.object,
    location: PropTypes.object,
    user: PropTypes.object,
    children: PropTypes.node,
    history: PropTypes.object,
    route: PropTypes.object,
    activeCommunity: PropTypes.string,
    globalModal: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
  };

  state = {
    openLoginModal: false
  };

  componentWillMount() {
    const { community } = this.props.auth;
    if (community && community !== 'home') {
      this.props.actions.setCommunity(community);
    }
  }

  componentDidMount() {
    const { actions, auth } = this.props;
    const { community } = auth;

    actions.setCommunity(community);
    actions.getCommunities();
    actions.getUser();
    actions.getEarnings('pending');

    if (auth.user && !auth.user.webOnboard.onboarding) {
      actions.showModal('onboarding');
      actions.webOnboard('onboarding');
    }
    // TODO do this after a timeout
    // window.addEventListener('focus', () => {
    //   if (this.props.newPosts)
    //   this.props.actions.refreshTab('discover');
    // });
  }

  componentDidUpdate(prevProps) {
    const { actions, auth, location, match, activeCommunity } = this.props;
    const { community } = match.params;
    if (community && activeCommunity !== community) {
      if (community === 'home') {
        this.props.history.push(`/${activeCommunity}/new`);
      } else actions.setCommunity(community);
    }

    if (location.pathname !== prevProps.location.pathname) {
      window.scrollTo(0, 0);
    }

    const userId = auth.user ? auth.user._id : null;
    const PrevUserId = prevProps.auth.user ? prevProps.auth.user._id : null;

    if (userId !== PrevUserId) {
      actions.userToSocket(userId);
    }

    if (!prevProps.auth.user && auth.user && !auth.user.webOnboard.onboarding) {
      actions.showModal('onboarding');
      actions.webOnboard('onboarding');
    }
  }

  toggleLogin() {
    this.setState({ openLoginModal: !this.state.openLoginModal });
  }

  closeModal() {
    this.props.history.push(this.props.location.pathname);
  }

  renderModal() {
    let { globalModal } = this.props;
    if (!globalModal) return null;
    globalModal = modals[globalModal] || globalModal;
    return (
      <Modal
        title={globalModal.title}
        header={globalModal.header}
        footer={globalModal.footer}
        close={() => this.props.actions.hideModal()}
        visible
      >
        {globalModal.body}
      </Modal>
    );
  }

  render() {
    const { location, user, match, children } = this.props;
    const temp = user && user.role === 'temp';
    const create = location.hash === '#newpost';
    const connectAccount = location.hash === '#connectAccount';

    let mobileEl = (
      <div className="mobileSplash">
        <h1>Relevant browser version doesn't currently support mobile devices</h1>
        <p>Please download a dedicated mobile app:</p>
        <p>
          <a
            href="https://itunes.apple.com/us/app/relevant-a-social-news-reader/id1173025051?mt=8"
            target="_blank"
          >
            <img alt="iOS App Store" src="https://relevant.community/img/appstore.png" />
          </a>
          &nbsp;&nbsp;&nbsp;&nbsp;
          <a
            href="https://play.google.com/store/apps/details?id=com.relevantnative&amp;pcampaignid=MKT-Other-global-all-co-prtnr-py-PartBadge-Mar2515-1"
            target="_blank"
          >
            <img
              alt="Google Play Store"
              src="https://relevant.community/img/googleplaystore.png"
            />
          </a>
        </p>
      </div>
    );

    let header;
    if (location.pathname === '/') {
      header = <Header match={match} toggleLogin={this.toggleLogin.bind(this)} />;
      mobileEl = null;
    }

    return (
      <div>
        <GlobalStyle />
        <EthTools>
          {header}
          <div style={{ display: 'flex', width: '100%' }}>
            {/* <CommunityNav {...this.props} /> */}
            {children}
          </div>
          <AuthContainer
            toggleLogin={this.toggleLogin.bind(this)}
            open={this.state.openLoginModal || temp}
            modal
            {...this.props}
          />
          <Eth.Consumer>
            {wallet => (
              <AddEthAddress
                connectAccount={connectAccount}
                closeModal={this.closeModal.bind(this)}
                {...this.props}
                {...wallet}
              />
            )}
          </Eth.Consumer>
        </EthTools>
        <Modal
          title="New Post"
          visible={create}
          className="createPostModal"
          close={() => this.props.history.push(location.pathname)}
        >
          <CreatePost modal />
        </Modal>
        {this.renderModal()}
        <ToastContainer />
        {mobileEl}
        {renderRoutes(this.props.route.routes)}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth.user,
  auth: state.auth,
  activeCommunity: state.community.active,
  navigation: state.navigation,
  globalModal: state.navigation.modal
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      ...navigationActions,
      ...authActions,
      getCommunities,
      getEarnings
    },
    dispatch
  )
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
