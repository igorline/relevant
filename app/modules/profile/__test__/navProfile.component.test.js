import React from 'react';
import renderer from 'react-test-renderer';
import { NavProfile } from 'modules/profile/navProfile.component';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
import { auth, earnings } from 'app/mockdata';

const props = {
  earnings: earnings.earnings,
  user: auth.auth.user
};

test('Snapshot NavProfile', () => {
  const tree = renderer
  .create(
    <MemoryRouter>
      <NavProfile {...props} />
    </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
