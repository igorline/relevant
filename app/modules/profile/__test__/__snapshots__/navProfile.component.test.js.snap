// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot NavProfile 1`] = `
.c0 {
  color: #0000ff;
}

.c0:hover * {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  color: #0000ff;
}

.c2 {
  color: #0000ff;
  color: #0000ff;
  -webkit-text-decoration-color: #0000ff;
  text-decoration-color: #0000ff;
}

.c2:hover * {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  color: #0000ff;
  margin-left: 0.5rem;
  color: #0000ff;
  -webkit-text-decoration-color: #0000ff;
  text-decoration-color: #0000ff;
}

.c3:hover * {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0px",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "borderBottomColor": "rgba(222,222,222,1.00)",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "1px",
      "borderLeftStyle": "solid",
      "borderRightStyle": "solid",
      "borderTopStyle": "solid",
      "display": "flex",
      "flexBasis": "0px",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0px",
      "position": "relative",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "display": "flex",
        "paddingBottom": "2rem",
        "paddingLeft": "1.5rem",
        "paddingRight": "1.5rem",
        "paddingTop": "1.5rem",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "justify",
          "WebkitFlexDirection": "row",
          "WebkitJustifyContent": "space-between",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "msFlexPack": "justify",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(0,0,0,1.00)",
            "fontFamily": "HelveticaNeue-Bold",
            "fontSize": "1.5rem",
            "lineHeight": "2.0999999999999996rem",
          }
        }
      >
        Alice
      </div>
      <a
        className="c0 ulink-mock"
        href="/user/wallet"
        hu={1}
        inline={1}
        onClick={[Function]}
        styles=""
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
              "color": "rgba(0,0,255,1.00)",
              "fontFamily": "HelveticaNeue",
              "fontSize": "0.75rem",
              "lineHeight": "1.0499999999999998rem",
              "textDecorationColor": "rgba(0,0,255,1.00)",
            }
          }
        >
          My Wallet
        </div>
      </a>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "1.5rem",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "display": "flex",
            "position": "relative",
          }
        }
      >
        <a
          className="c1 ulink-mock"
          href="/user/profile/alice"
          hu={0}
          inline={0}
          onClick={[Function]}
          styles=""
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitFlexBasis": "auto",
                "borderBottomLeftRadius": "2rem",
                "borderBottomRightRadius": "2rem",
                "borderTopLeftRadius": "2rem",
                "borderTopRightRadius": "2rem",
                "display": "flex",
                "flexBasis": "auto",
                "height": "4rem",
                "msFlexPreferredSize": "auto",
                "overflowX": "hidden",
                "overflowY": "hidden",
                "position": "relative",
                "width": "4rem",
                "zIndex": 0,
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.00)",
                  "backgroundImage": "url(\\"https://hips.hearstapps.com/elleuk.cdnds.net/15/37/2048x2730/2048x2730-e6-8278-458c32fb7c74-assets-elleuk-com-gallery-17063-taylor-swift-hairstyle-bob-getty-gallery-01-jpg.jpg\\")",
                  "backgroundPosition": "center",
                  "backgroundRepeat": "no-repeat",
                  "backgroundSize": "cover",
                  "bottom": "0px",
                  "height": "100%",
                  "left": "0px",
                  "position": "absolute",
                  "right": "0px",
                  "top": "0px",
                  "width": "100%",
                  "zIndex": -1,
                }
              }
            />
            <img
              alt=""
              className="css-accessibilityImage-9pa8cd"
              draggable={false}
              src="https://hips.hearstapps.com/elleuk.cdnds.net/15/37/2048x2730/2048x2730-e6-8278-458c32fb7c74-assets-elleuk-com-gallery-17063-taylor-swift-hairstyle-bob-getty-gallery-01-jpg.jpg"
            />
          </div>
        </a>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "vertical",
            "WebkitFlexDirection": "column",
            "WebkitFlexShrink": 1,
            "display": "flex",
            "flexDirection": "column",
            "flexShrink": 1,
            "msFlexDirection": "column",
            "msFlexNegative": 1,
            "paddingLeft": "1rem",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxDirection": "normal",
              "WebkitBoxOrient": "horizontal",
              "WebkitFlexDirection": "row",
              "display": "flex",
              "flexDirection": "row",
              "msFlexDirection": "row",
              "position": "relative",
            }
          }
        >
          <a
            className="c1 ulink-mock"
            href="/user/wallet"
            hu={0}
            inline={1}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              data-for="mainTooltip"
              data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"Earn Reputation by posting comments.\\\\nThe higher your score, the more weight your votes have.\\"}}"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitBoxAlign": "center",
                  "WebkitBoxDirection": "normal",
                  "WebkitBoxOrient": "horizontal",
                  "WebkitFlexDirection": "row",
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": "0.75rem",
                  "msFlexAlign": "center",
                  "msFlexDirection": "row",
                  "position": "relative",
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "WebkitFlexBasis": "auto",
                    "bottom": "0px",
                    "display": "flex",
                    "flexBasis": "auto",
                    "height": "1.5rem",
                    "marginRight": "0.15rem",
                    "msFlexPreferredSize": "auto",
                    "overflowX": "hidden",
                    "overflowY": "hidden",
                    "position": "relative",
                    "width": "1.5rem",
                    "zIndex": 0,
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n"
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0.00)",
                      "backgroundImage": "url(\\"r-emoji.png\\")",
                      "backgroundPosition": "center",
                      "backgroundRepeat": "no-repeat",
                      "backgroundSize": "contain",
                      "bottom": "0px",
                      "height": "100%",
                      "left": "0px",
                      "position": "absolute",
                      "right": "0px",
                      "top": "0px",
                      "width": "100%",
                      "zIndex": -1,
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="r-emoji.png"
                />
              </div>
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(0,0,0,1.00)",
                    "display": "flex",
                    "fontFamily": "HelveticaNeue-CondensedBold",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                    "position": "relative",
                  }
                }
              >
                11
              </div>
            </div>
          </a>
          <a
            className="c1 ulink-mock"
            href="/user/wallet"
            hu={0}
            inline={1}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              data-for="mainTooltip"
              data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"Get coins by upvoting quality links.\\\\nThe higher your Reputation the more coins you earn.\\"}}"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitBoxAlign": "center",
                  "WebkitBoxDirection": "normal",
                  "WebkitBoxOrient": "horizontal",
                  "WebkitFlexDirection": "row",
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": "0.75rem",
                  "msFlexAlign": "center",
                  "msFlexDirection": "row",
                  "position": "relative",
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "WebkitFlexBasis": "auto",
                    "bottom": "0px",
                    "display": "flex",
                    "flexBasis": "auto",
                    "height": "1.5rem",
                    "msFlexPreferredSize": "auto",
                    "overflowX": "hidden",
                    "overflowY": "hidden",
                    "position": "relative",
                    "width": "1.5750000000000002rem",
                    "zIndex": 0,
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n"
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0.00)",
                      "backgroundImage": "url(\\"relevantcoin.png\\")",
                      "backgroundPosition": "center",
                      "backgroundRepeat": "no-repeat",
                      "backgroundSize": "contain",
                      "bottom": "0px",
                      "height": "100%",
                      "left": "0px",
                      "position": "absolute",
                      "right": "0px",
                      "top": "0px",
                      "width": "100%",
                      "zIndex": -1,
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="relevantcoin.png"
                />
              </div>
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(0,0,0,1.00)",
                    "display": "flex",
                    "fontFamily": "HelveticaNeue-CondensedBold",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                    "marginLeft": "0.375rem",
                    "position": "relative",
                  }
                }
              >
                
                111
                
              </div>
            </div>
          </a>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxDirection": "normal",
              "WebkitBoxOrient": "horizontal",
              "WebkitFlexDirection": "row",
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginTop": "1rem",
              "msFlexAlign": "center",
              "msFlexDirection": "row",
              "position": "relative",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(128,128,128,1.00)",
                "fontFamily": "HelveticaNeue",
                "fontSize": "0.75rem",
                "lineHeight": "1.0499999999999998rem",
              }
            }
          >
            Pending Rewards: 
          </div>
          <div
            className="css-view-1dbjc4n"
            data-for="mainTooltip"
            data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"These are your projected earnings for upvoting quality posts.\\\\nRewards are paid out 3 days after a link is posted.\\"}}"
            style={
              Object {
                "WebkitAlignItems": "baseline",
                "WebkitBoxAlign": "baseline",
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "horizontal",
                "WebkitFlexDirection": "row",
                "alignItems": "baseline",
                "display": "flex",
                "flexDirection": "row",
                "marginRight": "0.75rem",
                "msFlexAlign": "baseline",
                "msFlexDirection": "row",
                "position": "relative",
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitFlexBasis": "auto",
                  "bottom": "-1px",
                  "display": "flex",
                  "flexBasis": "auto",
                  "height": "0.75rem",
                  "msFlexPreferredSize": "auto",
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": "0.7875000000000001rem",
                  "zIndex": 0,
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.00)",
                    "backgroundImage": "url(\\"relevantcoin.png\\")",
                    "backgroundPosition": "center",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "contain",
                    "bottom": "0px",
                    "height": "100%",
                    "left": "0px",
                    "position": "absolute",
                    "right": "0px",
                    "top": "0px",
                    "width": "100%",
                    "zIndex": -1,
                  }
                }
              />
              <img
                alt=""
                className="css-accessibilityImage-9pa8cd"
                draggable={false}
                src="relevantcoin.png"
              />
            </div>
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "WebkitTextDecorationColor": "rgba(0,0,0,1.00)",
                  "color": "rgba(0,0,0,1.00)",
                  "fontFamily": "HelveticaNeue",
                  "fontSize": "0.75rem",
                  "lineHeight": "1.0499999999999998rem",
                  "marginLeft": "0.1875rem",
                  "textDecorationColor": "rgba(0,0,0,1.00)",
                }
              }
            >
              
              120
              
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "baseline",
          "WebkitBoxAlign": "baseline",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "1.5rem",
          "msFlexAlign": "baseline",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <a
        c="#0000ff"
        className="c2 ulink-mock"
        href="/user/wallet"
        hu={1}
        inline={0}
        onClick={[Function]}
        styles=""
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
              "color": "rgba(0,0,255,1.00)",
              "fontFamily": "HelveticaNeue",
              "fontSize": "0.75rem",
              "lineHeight": "1.0499999999999998rem",
              "textDecorationColor": "rgba(0,0,255,1.00)",
            }
          }
        >
          Get Tokens
        </div>
      </a>
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "display": "flex",
            "position": "relative",
          }
        }
      >
            
      </div>
      <a
        c="#0000ff"
        className="c3 ulink-mock"
        href="/user/wallet"
        hu={1}
        inline={0}
        ml={1}
        onClick={[Function]}
        styles=""
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
              "color": "rgba(0,0,255,1.00)",
              "fontFamily": "HelveticaNeue",
              "fontSize": "0.75rem",
              "lineHeight": "1.0499999999999998rem",
              "textDecorationColor": "rgba(0,0,255,1.00)",
            }
          }
        >
          Invite Friends
        </div>
      </a>
    </div>
  </div>
</div>
`;
